# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit-test/tests/basic/delete-last-check-proto.js
# Commit: ec72f104edbf
# Full Hash: ec72f104edbfe88e22e2ebd778cc7890c21fd2ce
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 9 - Move proto from ObjectGroup to BaseShape. r=jonco
#   
#   The ObjectGroup still stores the proto as well but we don't really use it anymore.
#   This is simpler than removing the group's proto now because that would result in
#   "empty" groups and the group table then doesn't make sense anymore.
# ==============================================================================

diff -r cad616508f73 -r ec72f104edbf js/src/jit-test/tests/basic/delete-last-check-proto.js
--- /dev/null	Thu Jan 01 00:00:00 1970 +0000
+++ b/js/src/jit-test/tests/basic/delete-last-check-proto.js	Sat Mar 06 01:05:22 2021 +0000
@@ -0,0 +1,8 @@
+// Test that removing a non-dictionary property does not roll back the object's
+// shape to the previous one if the protos are different.
+
+var o = {x: 1};
+o.y = 2;
+Object.setPrototypeOf(o, null);
+delete o.y;
+assertEq(Object.getPrototypeOf(o), null);